pkg/groups/gsuite/grouper.go

```diff
@@ -135,7 +137,7 @@ func (g *Grouper) Members(group string) ([]rbacv1.Subject, error) {
 			return nil, errors.Wrapf(groups.ErrNotFound, "gsuite does not have group: %v", err)
 		}
 
-		return nil, err
+		return nil, errors.Wrapf(groups.ErrTemporary, "error retrieving group members: %v", err)
```

Comment:
Ok, looks like the destructuring logic is below, in `isNotFound`. Probably want to map the temp error to 5xx, but not 40[1,3]. Albeit, 403 can happen with a misconfiguration upstream. 400 is definitely not temporary.

Reply:
Ok, it might be best if we treat all errors as temporary, except 400 and 404. We can increase the set of terminal errors as they are discovered. The other error class we should handle here are timeouts, which can be detected by checking the cause against context.DeadlineExceeded or context.Cancelled.

Reply:
Updated.  PTAL.